### PR TITLE
Fix pagination

### PR DESCRIPTION
**Pull request by aderrod94**
There's a fixed problem with pagination logic for the different components: messages, user programs, all programs.

### CODE REVIEW

File: website/frontend/src/components/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -17,6 +17,15 @@ export const Pagination: VFC<Props> = ({ page, count, onPageChange, setShouldRel
...
   const isDisabledNext = page === totalPages || totalPages === 0;
+  const isAllProgramsPage = useRouteMatch(routes.allPrograms);
+  const isAllMessagesPage = useRouteMatch(routes.messages);
+  let url = routes.uploadedPrograms;
+
+  if (isAllProgramsPage) {
+    url = routes.allPrograms;
+  } else if (isAllMessagesPage) {
+    url = routes.messages;
+  }
```
Comment (nikitayutanov) [inline website/frontend/src/components/Pagination/Pagination.tsx:28]:
```suggestion
  const { pathname: url } = useLocation();
```

File: website/frontend/src/components/Pagination/Pagination.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import React, { VFC } from 'react';
-import { Link } from 'react-router-dom';
+import { Link, useRouteMatch } from 'react-router-dom';
```
Comment (nikitayutanov) [inline website/frontend/src/components/Pagination/Pagination.tsx:2]:
```suggestion
import { Link, useLocation } from 'react-router-dom';
```

**Review by nikitayutanov — COMMENTED** [review]
I guess there's more straightforward way to get current route

File: website/frontend/src/components/Pagination/Pagination.tsx (hunk 2 above)
Comment (aderrod94) [reply website/frontend/src/components/Pagination/Pagination.tsx:2]: 👍 thank you)

**Review by nikitayutanov — APPROVED** [review]

**Review by EugenWay — APPROVED** [review]

**Pull request merged**